pmlc/dialect/eltwise/tests/constant-fold.mlir

```diff
@@ -212,3 +212,28 @@ func @fold_div_f32_0(%arg0: f32) -> f32 {
   // CHECK-NEXT: %0 = "eltwise.div"(%arg0, %cst) : (f32, f32) -> tensor<f32>
   // CHECK-NEXT: return %0 : tensor<f32>
 }
+
+// Expected behavior in this case of cast is to fold
+// CHECK-LABEL: @fold_cast_si32
+func @fold_cast_si32(%arg0: tensor<3xsi32>) -> tensor<3xsi32> {
+  %0 = "eltwise.cast"(%arg0) : (tensor<3xsi32>) -> tensor<3xsi32>
+  return %0 : tensor<3xsi32>
+  // CHECK-NEXT: return %arg0 : tensor<3xsi32>
```

Comment:
Please don't use `%arg0` directly in a test:
https://mlir.llvm.org/getting_started/TestingGuide/#filecheck-best-practices

Use:
```
// CHECK-NEXT: return %{{.*}} : tensor<3xf32>
```